### PR TITLE
fix(settings): refactor change password and add form-handling to modal

### PR DESCRIPTION
**Pull request by escapedcat**
image
image
image

### CODE REVIEW

File: src/components/ConfirmModal.tsx [hunk 1]
```diff
@@ -38,21 +40,31 @@ export const ConfirmModal = ({
                 {headline}
               </ModalHeader>
 
-              {!!body && <ModalBody>{body}</ModalBody>}
+              {!!isFormModal ? (
```
Comment (escapedcat) [inline src/components/ConfirmModal.tsx:43]: Not sure if this is a good way to handle forms in modals

File: src/pages/Settings/ChangePwModal.tsx [hunk 2]
```diff
@@ -1,142 +1,155 @@
...
-    </ModalDialog>,
-    MODAL_ROOT,
+  return (
+    <>
+      <ConfirmModal
+        disclosure={confirmModal}
+        headline={t("settings.change_pw_a")}
+        isLoading={isLoading}
+        isFormModal={
+          <form onSubmit={handleSubmit(changePwHandler)}>
```
Comment (escapedcat) [inline src/pages/Settings/ChangePwModal.tsx:67]: I tried to define this part as a component further up but when validation kicks in it re-renders and fucks up the focus within the modal. Not sure how to solve this in a simple way. Tried a simple memo, but that didn't help.

File: src/components/ConfirmModal.tsx (hunk 1 above)
Comment (cstenglein) [inline src/components/ConfirmModal.tsx:43]:
```suggestion
              {isFormModal || (
```

File: src/pages/Settings/ChangePwModal.tsx (hunk 2 above)
Comment (cstenglein) [reply src/pages/Settings/ChangePwModal.tsx:67]: Lets keep it that way then.

**Review by cstenglein — APPROVED** [review]

**Comment by cstenglein** [comment]
Awesome stuff, thank you :)

**Pull request merged**